src/borg/remote.py

```diff
@@ -1141,31 +1159,56 @@ def get_many(self, keys, cache=True):
                 file = self.key_filename(key)
                 with open(file, 'rb') as fd:
                     self.hits += 1
-                    yield fd.read()
+                    yield self.unpack(fd.read())
             else:
                 for key_, data in repository_iterator:
                     if key_ == key:
-                        if cache:
-                            self.add_entry(key, data)
+                        transformed = self.add_entry(key, data, cache)
                         self.misses += 1
-                        yield data
+                        yield transformed
                         break
                 else:
                     # slow path: eviction during this get_many removed this key from the cache
                     t0 = time.perf_counter()
                     data = self.repository.get(key)
                     self.slow_lat += time.perf_counter() - t0
-                    if cache:
-                        self.add_entry(key, data)
+                    transformed = self.add_entry(key, data, cache)
                     self.slow_misses += 1
-                    yield data
+                    yield transformed
         # Consume any pending requests
         for _ in repository_iterator:
             pass
 
 
-def cache_if_remote(repository):
+def cache_if_remote(repository, *, decrypted_cache=False, pack=None, unpack=None, transform=None):
+    """
+    Return a Repository(No)Cache for *repository*.
+
+    If *decrypted_cache* is a key object, then get and get_many will return a tuple
+    (csize, plaintext) instead of the actual data in the repository. The cache will
+    store decrypted data, which increases CPU efficiency (by avoiding repeatedly decrypting
+    and more importantly MAC and ID checking cached objects).
+    Internally, objects are compressed with LZ4.
+    """
+    if decrypted_cache and (pack or unpack or transform):
+        raise ValueError('decrypted_cache and pack/unpack/transform are incompatible')
+    elif decrypted_cache:
+        key = decrypted_cache
+        cache_struct = struct.Struct('=I')
+        compressor = LZ4()
+
+        def pack(data):
```

Comment:
These are callables passed to Repository(No)Cache which defines this generic interface. I don't like changing parameter names in what is similar to a derived class.

Perhaps reorder them transform-pack-unpack, so that the relation is more clear. And a comment or so.